lib/api/core/notifier.js

```diff
@@ -272,13 +272,39 @@ class NotifierController {
           this.kuzzle.notifier.notify(cachedRooms, request, notification);
           return this.kuzzle.services.list.internalCache.remove(this.cacheKeyPrefix + id);
         })
-        .asCallback(callback);
-    }, error => {
-      if (error) {
-        this.kuzzle.pluginsManager.trigger('log:error', error);
-        return false;
-      }
-    });
+        .then(()=> {
```

Comment:
These 2 `then` blocks overlap in functionality. Refer to the conversation about dropping the internal cache.